app/flatpak-builtins-install.c

```diff
@@ -299,10 +300,15 @@ flatpak_builtin_install (int argc, char **argv, GCancellable *cancellable, GErro
   if (!flatpak_split_partial_ref_arg (name, &opt_arch, &branch, error))
     return FALSE;
 
+  if (branch != NULL && *branch != 0)
+    actual_branch = g_strdup (branch);
+  else
+    actual_branch = flatpak_dir_get_remote_default_branch (dir, repository);
+
   if (!opt_app && !opt_runtime)
     opt_app = opt_runtime = TRUE;
 
-  ref = flatpak_dir_find_remote_ref (dir, repository, name, branch, opt_arch,
+  ref = flatpak_dir_find_remote_ref (dir, repository, name, actual_branch, opt_arch,
                                      opt_app, opt_runtime, &is_app, cancellable, error);
```

Comment:
I don't think this is quite right. flatpak_dir_find_remote_ref() will look for all the existing matches, and if you don't specify a branch, but there is only one available then will pick that, independent of the default branch. If there are multiple matches then it return an error with the available branches and let the user pick. What we want to do is in the later case, *if* the default branch is among the possible branches, then pick that, otherwise error out as before.

So, you need to pass a default branch argument to flatpak_dir_find_remote_ref()